sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionManager.java

```diff
@@ -235,8 +235,11 @@ public void close() {
         final List<Mono<Void>> closeables = sessionReceivers.values().stream()
             .map(receiver -> receiver.closeAsync())
             .collect(Collectors.toList());
-
-        Mono.when(closeables).block(operationTimeout);
+        
+        if(!closeables.isEmpty()){
+            Mono.when(closeables).block(operationTimeout);
+        }
+        
```

Comment:
its a mitigation, a little improvement, but it doesnt solve the main issue - "calling blocking methods within nonblocking context"